grammar.js

```diff
@@ -287,11 +287,7 @@ module.exports = grammar({
 
     for: $ => seq(
       'for',
-      commaSep1(field('pattern', choice(
-        $._lhs,
-        $.rest_assignment,
-        $.destructured_left_assignment
-      ))),
+      field('pattern', choice($._lhs, $.left_assignment_list)),
```

Comment:
left_assignment_list already includes _lhs as far as I can tell, so this could be simplified to

 
```suggestion
      field('pattern', $.left_assignment_list),
```

Reply:
I think the current structure is good. Right now, we only want to create a `left_assignment_list` node if there are either
1. multiple patterns
2. a trailing comma

This is useful because, in these circumstances, the left-hand-side is interpreted as *destructuring* the right-hand side, so that this holds:

```ruby
a, b = [1, 2]
assert_equal a, 1

c, = [1, 2]
assert_equal c, 1
```

Reply:
That's exactly why I did it that way - it's also useful for our use case to be able to avoid the extra node for the simple, non-destructuring case.

@maxbrunsfeld - I see you've approved, but could you also merge when you're ready?